### ISSUE TITLE
Adding attribute about "degree of uncertainty"

### ISSUE THREAD

**Issue by damianooldoni** [trias-project/indicators#44]
From e-mail of @amyjsdavis:

> By the way, Diederik and I are very enthusiastic about the data cube and your approach to handling spatial uncertainty. We ask that you consider adding one attribute that identifies whether the grid cell contains at least one presence with an "exact" location, or whether the presences in the grid cell have all been randomly assigned. This attribute will be very useful to our modeling if it turns out that we need to reduce the amount of the uncertainty in our model.

**Comment by damianooldoni** [comment]
I understand your and @DiederikStrubbe's concern. And I agree about the idea of adding a binary label specifying the "degree of uncertainty" linked to each grid cell. 

This is my suggestion: adding a column `n_low_unc` which tells us how many occurrences have an uncertainty less than 500 meter (low uncecrtainty). Something like this I have already developed:

species | CELLCODE | year |  n | n_low_unc
-- | -- | -- | -- | --
Branta canadensis | 1kmE3922N3105 | 2014 | 31 | 28
Branta canadensis | 1kmE3922N3105 | 2014 | 24 | 24
... | ... | ... | ... | ...
Branta canadensis | 1kmE3922N3110 | 2018 | 2 | 1

The idea we thought during the meeting in January of not reassigning the point if uncertainty < 500 m is nothing else than a trick to speed up the data processing, an approximation, a workaround which actually I would like to remove by parallel computing and cloud computing... and the reason is quite simple: it doesn't exist an **exact** location of an observation as it doesn't exist any exact measurement at all! Everything we measure has an uncertainty. The values in `decimalLatitude` and `decimalLongitude` are not more likely to be exact than others within the given coordinate uncertainty (unless I describe a density of probability function of the error, which is almost never included in basic error theory). If I say: "_this table is 2m ± 0.05 long_" then I admit that this table can be one of the infinite real values in [1.95, 2.05]. The value 2m has no any privileged role and it is as likely as any other value.

Here below justa preview of an occurrence point where my idea can be fully understood:
![image](https://user-images.githubusercontent.com/33662631/52219248-8ac9c900-289c-11e9-8352-9dafe00d47b4.png)
if we zoom:
![image](https://user-images.githubusercontent.com/33662631/52219358-c5cbfc80-289c-11e9-981a-4ee8a5c589ab.png)

The green point (decimalLatitude and decimalLongitude) is not more true than the reassigned red one. What is true is the circle!!! In both cases (red or green point) we assign the point to one and only one cell, this the most important thing. 

So, @amyjsdavis  and @DiederikStrubbe : I think my solution can provide, maybe reformulated in other words, your will of storing how many points actually are "good" points :smile:

**Comment by peterdesmet** [comment]
Thanks Damiano!

**Comment by amyjsdavis** [comment]
@damianooldoni : I should not have use the word "exact". Indeed, no spatial location is "exact", but with points of small uncertainty, we can be reasonably sure that they are indeed located within a 1km grid cell, as compared to point with a larger uncertainty. We can talk about this more at the meeting today.

**Comment by damianooldoni** [comment]
Thanks @amyjsdavis for your ideas. Based on what we discussed in the TrIAS core meeting this morning, we can agree on this strategy:

1. assign the occurrence point within circle of uncertainty
2. for each cell, get the least uncertainty related to the occurrence points assigned to the cell

This strategy is interesting because for occupancy or risk assessment we work in a binary way (YES, occupied; NO, not occupied). This way to work will preserve would preserve a detailed information about the degree of uncertainty than what I wrote (https://github.com/trias-project/pipeline/issues/44#issuecomment-460299999, with a fixed threshold, 500m). We can for example see differences in model results (or occupancy) by varying the threshold used on this attribute.
